tests/assembleCoaddTestUtils.py

```diff
@@ -0,0 +1,625 @@
+        lsstLat = -30.244639*u.degree
+        lsstLon = -70.749417*u.degree
+        lsstAlt = 2663.*u.m
+        lsstTemperature = 20.*u.Celsius
+        lsstHumidity = 40.  # in percent
+        lsstPressure = 73892.*u.pascal
+        loc = EarthLocation(lat=lsstLat,
+                            lon=lsstLon,
+                            height=lsstAlt)
+
+        time = Time(2000.0, format="jyear", scale="tt")
+        if randomizeTime:
+            # Pick a random time within a 6 hour window
+            time += 6*u.hour*(self.rngMods.random() - 0.5)
+        radec = SkyCoord(dec=self.dec.asDegrees(), ra=self.ra.asDegrees(),
+                         unit='deg', obstime=time, frame='icrs', location=loc)
+        airmass = float(1.0/np.sin(radec.altaz.alt))
+        obsInfo = makeObservationInfo(location=loc,
+                                      detector_exposure_id=exposureId,
+                                      datetime_begin=time,
+                                      datetime_end=time,
+                                      boresight_airmass=airmass,
+                                      boresight_rotation_angle=Angle(0.*u.degree),
+                                      boresight_rotation_coord='sky',
+                                      temperature=lsstTemperature,
+                                      pressure=lsstPressure,
+                                      relative_humidity=lsstHumidity,
+                                      tracking_radec=radec,
+                                      altaz_begin=radec.altaz,
+                                      observation_type='science',
+                                      )
+        visitInfo = MakeRawVisitInfoViaObsInfo.observationInfo2visitInfo(obsInfo)
+        return visitInfo
+
+    def defineFilter(self, lambdaEff=476.31, lambdaMin=405., lambdaMax=552., filterName="gTest"):
+        """Construct a `Filter` with sufficient information to calculate DCR.
+
+        Parameters
+        ----------
+        lambdaEff : `float`, optional
+            The effective wavelength of the filter.
+        lambdaMin : float, optional
+            The minimum wavelength of the filter with greater than 1% transmission.
+        lambdaMax : float, optional
+            The maximum wavelength of the filter with greater than 1% transmission.
+        filterName : `str`, optional
+            The simplified name of the filter.
+
+        Returns
+        -------
+        filterInfo : `lsst.afw.image.Filter`
+            The filter definition.
+        """
+        # Use LSST g band values for the test.
+        afwImageUtils.defineFilter(filterName, lambdaEff,
+                                   lambdaMin=lambdaMin, lambdaMax=lambdaMax)
+        return afwImage.Filter(filterName)
+
+    def makeTestImage(self, expId, noiseLevel=None, psfSize=None, backgroundLevel=None,
+                      detectionSigma=5.):
+        """Make a reproduceable PSF-convolved masked image for testing.
```

Comment:
Good point. This doesn't actually need to modify the class in place -- it can just return the images and the test case can build the `dict` of exposures.